include/logger.h

```diff
@@ -8,14 +8,22 @@
 
 // * Log record : each line of the log
 // * log format : tuple like <Date, Name, Message>
-// * log entity : Date, Name, Message
+// * log property : Date, Name, Message
 #define LOG(severity)                                                          \
   logger::Logger::getInstance().log<logger::severity>(                         \
-      {__func__, __FILE__, __LINE__})
+      {__FILE__, __func__, __LINE__})
+
+
+std::string hex(intptr_t v) {
+  std::stringstream ss;
+  ss << std::hex << "0x" << v;
+  return ss.str();
+}
 
 namespace logger {
 
 // TODO: how to specify THRESHOLD in terms of these? make two header files? NO!
+// TODO: JEB suggested we should put these in the log config
```

Comment:
the best idea i could come up with is:
```
#include <clayer.h>          // defines everything needed in logconfig.h or clayerconfig.h
#include "clayerconfig.h"       // defines everything needed in logger.h and analyser.h
#include <logger.h>            // template meta programming here
```

as you could guess, this may not be generalisable, in that there might be circular dependencies in clayer.h and clayer_config.h but I feel they can be reduced using better design decisions. In the end, we may want to either prove that clayer.h is required or remove it.